crates/control/src/search_suggestor.rs

```diff
@@ -0,0 +1,175 @@
+use color_eyre::Result;
+use context_attribute::context;
+use framework::{AdditionalOutput, MainOutput};
+use nalgebra::{DMatrix, Isometry2, Point2};
+use serde::{Deserialize, Serialize};
+use types::{
+    ball_position::{BallPosition, HypotheticalBallPosition},
+    field_dimensions::FieldDimensions,
+    parameters::SearchSuggestorParameters,
+};
+
+#[derive(Deserialize, Serialize)]
+pub struct SearchSuggestor {
+    heatmap_dimensions: (usize, usize),
+    heatmap: DMatrix<f32>,
+}
+
+#[context]
+pub struct CreationContext {
+    field_dimensions: Parameter<FieldDimensions, "field_dimensions">,
+    search_suggestor_configuration: Parameter<SearchSuggestorParameters, "search_suggestor">,
+}
+
+#[context]
+pub struct CycleContext {
+    search_suggestor_configuration: Parameter<SearchSuggestorParameters, "search_suggestor">,
+    ball_position: Input<Option<BallPosition>, "ball_position?">,
+    invalid_ball_positions: Input<Vec<HypotheticalBallPosition>, "invalid_ball_positions">,
+    robot_to_field: Input<Option<Isometry2<f32>>, "robot_to_field?">,
+    field_dimensions: Parameter<FieldDimensions, "field_dimensions">,
+    heatmap: AdditionalOutput<DMatrix<f32>, "ball_search_heatmap">,
+}
+
+#[context]
+#[derive(Default)]
+pub struct MainOutputs {
+    pub suggested_search_position: MainOutput<Option<Point2<f32>>>,
+}
+
+impl SearchSuggestor {
+    pub fn new(_context: CreationContext) -> Result<Self> {
+        let heatmap_dimensions = (
+            _context.field_dimensions.length.round() as usize
+                * _context.search_suggestor_configuration.cells_per_meter,
+            _context.field_dimensions.width.round() as usize
+                * _context.search_suggestor_configuration.cells_per_meter,
```

Comment:
First multiply, then round. This makes the result more correct.
If field-length were 2.4 and you had 10 cells per meter, you would calculate around 20 cells whereas (2.4 * 10).round() returns 24 cells which is what was specified in the config

Reply:
Done